tests/sdk/test_config.py

```diff
@@ -32,6 +34,49 @@ def test_add_load_profile(tmpdir):
     assert profile_1 == profile_2
 
 
+def test_add_load_profile_from_file(tmpdir):
+    path = tmpdir / 'substra.cfg'
+    conf = {
+       "node-1": {
+           "auth": {
+               "username": "node-1"
+           },
+           "insecure": False,
+           "url": "http://substra-backend.node-1.com",
+           "version": "0.0"
+       },
+    }
+
+    path.write_text(json.dumps(conf), "UTF-8")
+    keyring.set_password('node-1', conf['node-1']['auth']['username'], 'foo')
+    manager = configuration.Manager(str(path))
+    profile = manager.load_profile('node-1')
+
+    assert conf['node-1']['auth']['username'] == profile['auth']['username']
+    assert keyring.get_password(service_name='node-1', username='node-1') == 'foo'
+
+
+def test_add_load_bad_profile_from_file(tmpdir):
+    path = tmpdir / 'substra.cfg'
+    conf = {
+       "node-1": {
+           "auth": {
+               "username": "node-1"
+           },
+           "insecure": False,
+           "url": "http://substra-backend.node-1.com",
+           "version": "0.0"
+       },
+    }
+
+    path.write_text(json.dumps(conf), "UTF-8")
+    keyring.set_password('node-1', conf['node-1']['auth']['username'], 'foo')
+    manager = configuration.Manager(str(path))
+
+    with pytest.raises(configuration.ProfileNotFoundError):
+        manager.load_profile('foo')
```

Comment:
Since the only test is about the error raised when trying to access a non-existing profile, I don't think that loading profiles from a file make any difference from loading the profiles with a python object. It's therefore redundant with the test on L89-L90.

You could however try to write a bad configuration to the file and see what error is being raised by `configuration.Manager(str(path))`

Reply:
Yes, I'll do it 🙂